### ISSUE TITLE
Black screen on startup, Vive

### ISSUE THREAD

**Issue by Vaellian1020** [KozGit/DOOM-3-BFG-VR#120]
I installed the mod via the installer on a clean copy of Doom3

i started DoombfgVR and just nothing happens, i get a black screen, my vive is tracking and i can see my chaperone and stuff, but nothing beyond that

**Comment by lodanap** [comment]
I have exactly the same problem on my HTC Vive. I'm using the latest 17.3.1 AMD drivers and the non beta version of SteamVR.

**Comment by ColJay** [comment]
I'm also getting this. 

I've an AMD RX480 and a Vive. I have run with 17.3.1 drives and also down graded to the 17.2.1. Same symptoms.

**Comment by Dougiebabe** [comment]
Also getting the same issue. AMD RX470

New Install of Doom3:BFG - no mods other than VR mod.
Opens Steam VR if not already open fine
Opens new window 'Doom 3 BFG VR: Full Possessed'
With window 'active' I can hear sounds but just get a black screen
Moving mouse I can hear what sounds like the menu items being scrolled on

While typing this I must have managed to click on the right button and am now in game although my height is broken probably due to the headset being on the floor.

Seems like the menu is the bit that is broken?

**Comment by Vaellian1020** [comment]
I'll have to try that, not sure if it's an AMD think, but i have a Fury X and a Ryzen. Im on latest drivers 

The height is Janky but i think there's a console command for that

**Comment by CarlKenner** [comment]
I think this is an ATI (aka AMD) issue.  But since we don't have ATI cards, it's impossible for us to replicate or fix.

**Comment by chester0** [comment]
same problem on the Vive, 17.3.1 AMD drivers, beta version of SteamVR, managed to get into with random buttons but could not move or do anything after

**Comment by Dougiebabe** [comment]
I would agree with Carl that this seems limited to AMD cards only as it seems everyone has these cards who is experiencing issues.

Just to reiterate how I managed to get it working:

Install Doom 3
Install Mod using exe
Run Application from icon on desktop (Fully Possessed )
If you do not hear menu music/sounds click on window that appears, then I pressed the windows key but I would imagine any key will work.
Once you hear menu music do not move mouse or press any other buttons other than enter 2 or 3 times

WARNING! If you do not have the headset on your head the height is totally messed up, this is easier to start if you have 2 people otherwise try and stand as upright as you normally would be with headset on head and then press enter. Wireless K/B would also help here if one person.

This should get you into game, once in I have found most things work fine.

**Comment by KozGit** [comment]
I don't have an AMD card to test with, so it's difficult to investigate this - but I've attached a test build here to check the first thing that comes to mind regarding this issue - if anyone feels like testing, just extract the .exe from the .zip, drop it in your bfg folder and run it.  Let me know if it has any effect.

[Doom3BFGVR_amdTest.zip](https://github.com/KozGit/DOOM-3-BFG-VR/files/836978/Doom3BFGVR_amdTest.zip)

Also, if you start the game without the headset and your height is wrong, just use the recenter hmd command ( either bind a key/button or use the voice command ) to reset the hmd origin.

**Comment by Dougiebabe** [comment]
I just tried, still getting the black menu screen unfortunately.

I wasn't actually able to play this evening when trying to show a friend, it got to the main menu, normal enter 3 times, steam 'now loading' screen and then windows crash. This .exe has fixed this problem for me and the game now loaded. 

I'm wondering if that issue was due to me playing the vanilla game earlier though.

**Comment by CarlKenner** [comment]
I think the vanilla game interferes a bit, because it searches the vanilla game's save folder and configuration files after looking in the Fully Possessed save folder. That's my fault, sorry. We might have to include empty .cfg files that match the names of the vanilla game's .cfg files and overrride them.

**Comment by CarlKenner** [comment]
This is a duplicate of issue #9.

**Comment by KozGit** [comment]
I've got this sorted out. Testing positional tracking in the menus, will update when done.

**Comment by KozGit** [comment]
I have to say, the performance using a Vive with an RX480 is not very good unfortunately. Rendering the interactions just kills performance in certain instances.  Even with MSAA disabled and vr_pixelDensity set to 1 it's a problem -  AMD/Valve really need to get aysnc reprojection working in SteamVR.   ( And we should probably start trying to figure out what's killing performance in certain scenes )

**Comment by Dougiebabe** [comment]
Having not used a nVidia card I cannot comment on the performance but as no one has ever complained I can only assume that it's fine where as the AMD cards do seem to have some chronic dipping at times, even when the scenes don't look too intensive. Is it just whenever there there are 'interactions' as you call them?

Thank you for looking at this issue though 👍

**Comment by KozGit** [comment]
Allright, should have this sorted out now.  Added basic positional tracking to the menu screens as well.  I've attached another test .exe to the bottom of this message, let me know how it works.

> AMD cards do seem to have some chronic dipping at times, even when the scenes don't look too intensive. Is it just whenever there there are 'interactions' as you call them?

Yes, by interactions I mean light interactions - when the engine renders the scene and applies dynamic lighting.  For instance in the very first level when you are in the hangar, looking at the dropship causes a big hit in performance.  It happens on nVidia cards as well, it just seems much more pronounced on the Rx480 I just tested with than the nVidia cards I've used.  When you couple that with AMDs lack of asynchronous warp on the Vive, it's really noticeable.   At least AMD has said async for the Vive should be out soon.


[Doom3BFGVR_AMD_Test2.zip](https://github.com/KozGit/DOOM-3-BFG-VR/files/852709/Doom3BFGVR_AMD_Test2.zip)

**Comment by ColJay** [comment]
Hi,

I can confirm this fixed the menu issue for me, i was able to see and navigate into the game. The positioning in the cut-scene at the beginning was a little off, but once is game it was bang on. 

Only issue i now have is the Vive controllers don't appear to do anything, although this is more likely down to some of the tweaks i've been trying

Thanks for looking at this!

**Comment by Dougiebabe** [comment]
Also can confirm this is now working for me with the new exe.

I had always started further in for some reason when I couldn't see the main menu so now have started from the beginning and can confirm the frame drops do seem quite often whenever there is something going on.

I've also learnt something about ASW, I had no idea it was out for Vive but only on nVidia cards! Learn something new every day!

Thanks again!

**Comment by KozGit** [comment]
Glad it's working for you!  I'm going to leave this issue open until 022 is released in case anyone else is having the same problem.

ColJay, if you are still having issues with the controllers tracking, you can open a new issue, I'd like to keep this one focused on the black screen issue.   ( Make sure you don't have a gamepad plugged in somewhere with a button or stick being pressed on it. )
